UnityProject/Assets/Scripts/Systems/ChemistryEffects/ChemExplosionPullPush.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using System.Collections;
+using UnityEngine;
+using Systems.Explosions;
+using HealthV2;
+using Core;
+using Cysharp.Threading.Tasks;
+using Logs;
+using Systems.Atmospherics;
+using TileManagement;
+using UniversalObjectPhysics = Core.Physics.UniversalObjectPhysics;
+
+namespace Chemistry.Effects
+{
+	[CreateAssetMenu(fileName = "effect", menuName = "ScriptableObjects/Chemistry/Effect/ChemExplosionPullPush")]
+	public class ChemExplosionPullPush : ChemExplosion
+	{
+		[SerializeField, Tooltip("Should this effects force push not pull")] private bool ShouldPush = false;
+		[SerializeField] private string effectName = "Dark Matter";
+
+
+		[SerializeField] private GameObject cataclysmPrefab = null;
+		[SerializeField] private float cataclysmThreshold = 2000;
+
+		public override IEnumerator NowExplosion(MonoBehaviour sender, float amount)
+		{
+			yield return WaitFor.Seconds(Delay);
+
+			UniversalObjectPhysics objectBehaviour = sender.GetComponentCustom<UniversalObjectPhysics>();
+			RegisterObject registerObject = sender.GetComponentCustom<RegisterObject>();
+			BodyPart bodyPart = sender.GetComponentCustom<BodyPart>();
+			ExplosionNode node = ExplosionTypes.NodeTypes[explosionType];
+
+			float strength = ChemistryUtils.CalculateYieldFromReaction(amount, potency);
+
+			bool insideBody = bodyPart == true && bodyPart.HealthMaster == true;
+
+			var picked = sender.GetComponentCustom<Pickupable>();
+			if (picked != null)
+			{
+				//If sender is in an inventory use the position of the inventory.
+				if (picked.ItemSlot != null)
+				{
+					objectBehaviour = picked.ItemSlot.ItemStorage.GetRootStorageOrPlayer().GetComponentCustom<UniversalObjectPhysics>();
+					registerObject = picked.ItemSlot.ItemStorage.GetRootStorageOrPlayer().GetComponentCustom<RegisterObject>();
+				}
+			}
+
+			if (strength <= 0) yield break;
+
+			strength = ShouldPush ? -strength : strength;
+			Vector3 positionOfExplosion;
+
+			if (registerObject == null)
+			{
+				if (insideBody) positionOfExplosion = bodyPart.HealthMaster.RegisterTile.WorldPosition;
+				else positionOfExplosion = objectBehaviour.registerTile.WorldPosition;
+			}
+			else positionOfExplosion = registerObject.WorldPosition;
+
+			Explosion.StartExplosion(positionOfExplosion.CutToInt(), strength, node, stunNearbyPlayers: false, radiusMultiplier: 10);
```

Comment:
CutToInt -> round toint,   CutToInt  is from the days when we had trouble converting from local to worldand we didn't know what we were doing

Reply:
(behaviour differences 
cut = 0.1 = 0 , 0.9 = 0
Round = 0.1 = 0, 0.9 = 1
)